src/components/NewsletterSignup/NewsletterSignup.stories.tsx

```diff
@@ -112,6 +113,57 @@ export const WithControls: Story = {
   },
   render: (args) => <NewsletterSignup {...args} />,
 };
+let counter = 0;
+
+function NewsletterSignupOnSubmitExampleComponent() {
+  const [view, setView]: [
+    NewsletterSignupViewType,
+    React.Dispatch<React.SetStateAction<NewsletterSignupViewType>>
+  ] = React.useState("form");
```

Comment:
This is interesting. No need to remove, just pointing out I'd normally do `const [...] = useState<NewsletterSignupViewType>("form");` but this is nice to see.